packages/grafana-runtime/src/services/backendSrv.ts

```diff
@@ -15,29 +21,95 @@ export type BackendSrvRequest = {
    * Please have a look at {@link https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API | Fetch API}
    * for supported headers.
    */
-  headers?: any;
+  headers?: Record<string, any>;
 
   /**
    * HTTP verb to perform in the remote call GET, POST, PUT etc.
    */
   method?: string;
 
   /**
-   * If set to true an alert with the response message will be displayed
-   * upon successful remote call
+   * Set to false an success application alert box will not be shown for successful PUT, DELETE, POST requests
    */
   showSuccessAlert?: boolean;
 
+  /**
+   * Set to false to not show an application alert box for request errors
+   */
+  showErrorAlert?: boolean;
+
   /**
    * Provided by the initiator to identify a particular remote call. An example
    * of this is when a datasource plugin triggers a query. If the request id already
    * exist the backendSrv will try to cancel and replace the previous call with the
    * new one.
    */
   requestId?: string;
-  [key: string]: any;
+
+  /**
+   * Set to to true to not include call in query inspector
+   */
+  silent?: boolean;
+
+  /**
+   * The data to send
+   */
+  data?: any;
+
+  /**
+   * Query params
+   */
+  params?: Record<string, any>;
+
+  /**
+   * Indicates whether or not cross-site Access-Control requests should be made using credentials such as cookies, authorization headers or TLS client certificates. Setting withCredentials has no effect on same-site requests.
+   * In addition, this flag is also used to indicate when cookies are to be ignored in the response.
+   */
+  withCredentials?: boolean;
 };
 
+/**
+ * Response for fetch function in {@link BackendSrv}
+ *
+ * @public
+ */
+export interface FetchResponse<T = any> {
```

Comment:
This should probably extend `Response` instead of redeclaring its props.

Reply:
yea, think your right, but we are not returning the raw Response yet (it has a bunch more), so we can't do that right now without more refactoring and updating tests, but it's a good idea